tests/integration/mixer/telemetry/logs/accesslog_test.go

```diff
@@ -0,0 +1,154 @@
+	ingInst           *ingress.Instance
+)
+
+func TestIstioAccessLog(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			_, g, ing := setupComponentsOrFail(t)
+
+			ns := namespace.ClaimOrFail(t, ctx, ist.Settings().SystemNamespace)
+			g.ApplyConfigOrFail(
+				t,
+				ns,
+				bookinfo.TelemetryLogEntry.LoadOrFail(t))
+			defer g.DeleteConfigOrFail(
+				t,
+				ns,
+				bookinfo.TelemetryLogEntry.LoadOrFail(t))
+
+			util.AllowRuleSync(t)
+
+			err := util.VisitProductPage(ing, 30*time.Second, 200, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page: %v", err)
+			}
+
+			util.GetAndValidateAccessLog(ns, t, "istio-mixer-type=telemetry", "mixer",
+				validateLog)
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("mixer_telemetry_logs", m).
+		RequireEnvironment(environment.Kube).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, nil)).
+		Setup(testsetup).
+		Run()
+}
+
+func testsetup(ctx resource.Context) error {
+	bookinfoNs, err := namespace.New(ctx, "istio-bookinfo", true)
+	if err != nil {
+		return err
+	}
+	bookinfoNamespace = &bookinfoNs
+	if _, err := bookinfo.Deploy(ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookInfo}); err != nil {
+		return err
+	}
+	g, err := galley.New(ctx, galley.Config{})
+	if err != nil {
+		return err
+	}
+	galInst = &g
+	if _, err = mixer.New(ctx, mixer.Config{Galley: g}); err != nil {
+		return err
+	}
+	ing, err := ingress.New(ctx, ingress.Config{Istio: ist})
+	if err != nil {
+		return err
+	}
+	ingInst = &ing
+
+	bookinfoGateWayConfig, err := bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespace(bookinfoNs.Name())
+	if err != nil {
+		return err
+	}
+	err = g.ApplyConfig(bookinfoNs, bookinfoGateWayConfig)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setupComponentsOrFail(t *testing.T) (bookinfoNs namespace.Instance, g galley.Instance,
+	ing ingress.Instance) {
+	if bookinfoNamespace == nil {
+		t.Fatalf("bookinfo namespace not allocated in setup")
+	}
+	bookinfoNs = *bookinfoNamespace
+	if galInst == nil {
+		t.Fatalf("galley not setup")
+	}
+	g = *galInst
+	if ingInst == nil {
+		t.Fatalf("ingress not setup")
+	}
+	ing = *ingInst
+	return
+}
+
+func validateLog(content string) error {
+	if !strings.Contains(content, "newlog") {
+		return fmt.Errorf("accesslog doesnt contain newlog instance. Log %v", content)
+	}
+	if !strings.Contains(content, "\"level\":\"warn\"") {
+		return fmt.Errorf("accesslog doesnt contain warning level. Log %v", content)
+	}
+	if !strings.Contains(content, "\"destination\":\"details\"") || !strings.Contains(content,
+		"\"destination\":\"reviews\"") || !strings.Contains(content, "\"destination\":\"productpage\"") {
+		return fmt.Errorf(
+			"accesslog doesnt contain either details or reviews or productpage destination. Log %v", content)
+	}
+	if !strings.Contains(content, "\"responseCode\":200") {
```

Comment:
should we think about relaxing this?  do we care about what the response code is?  or just merely that the code has been populated?

Reply:
removed 200 code requirement... agreed just having responseCode for this particular test is enough